elyra/metadata/runtime_image.py

```diff
@@ -0,0 +1,228 @@
+
+    replace = Bool(False, help='Replace existing runtime image metadata with this instance.', config=True)
+
+    name = Unicode(None, config=True, allow_none=True,
+                   help="The canonical name of this kfp runtime.  Must be lowercase alphanumeric, beginning with "
+                        "alpha and can include embedded hyphens ('-') and underscores ('_'). Required.")
+
+    display_name = Unicode(None, config=True, allow_none=True,
+                           help="The display name of this runtime image. Required.")
+
+    image_name = Unicode(None, config=True, allow_none=True,
+                         help="The image name, including organization, name e tag. Required.")
+
+    metadata_namespace = RuntimeImage.namespace
+    metadata_manager = Instance(MetadataManager)
+
+    def _metadata_manager_default(self):
+        return MetadataManager(namespace=self.metadata_namespace)
+
+    aliases = {
+        'name': 'Image.name',
+        'display_name': 'Image.display_name',
+        'image_name': 'Image.image_name'
+    }
+
+    flags = {'replace': ({'Image': {'replace': True}}, "Replace existing runtime metadata with this instance."),
+             'debug': base_flags['debug'],
+             }
+
+    def start(self):
+        self._validate_parameters()
+
+        # init with required, conditionally add optional  # TODO - drive from metadata?  Will need better
+        metadata = dict(
+            image_name=self.image_name
+        )
+
+        runtime_image = RuntimeImage(schema_name=self.schema_name, name=self.name,
+                                     display_name=self.display_name, metadata=metadata)
+
+        ex_msg = None
+        resource = None
+        try:
+            resource = self.metadata_manager.add(self.name, runtime_image, replace=self.replace)
+        except Exception as ex:
+            ex_msg = str(ex)
+
+        if resource:
+            print("Metadata for {} runtime image '{}' has been written to: {}"
+                  .format(self.schema_name, self.name, resource))
+        else:
+            if ex_msg:
+                self.log_and_exit("The following exception occurred while saving metadata '{}' for {} runtime image: {}"
+                                  .format(self.name, self.schema_name, ex_msg), display_help=True)
+            else:
+                self.log_and_exit("A failure occurred while saving metadata '{}' for {} runtime.  Check log output."
+                                  .format(self.name, self.schema_name), display_help=True)
+
+    def _validate_parameters(self):
+        self.confirm_required("name", self.name)
+        self.confirm_required("display_name", self.display_name)
+        self.confirm_required("image_name", self.image_name)
+
+
+class InstallRuntimeImages(AppUtilMixin, Application):
+    version = __version__
+    description = """Install runtime image metadata."""
+
+    subcommands = Dict({
+        'image': (Image, Image.description.splitlines()[0])
```

Comment:
Is there a reason we need the `image` subcommand?  Unless we expect other schemas relative to the runtime-images namespace, I think this should be removed.

Reply:
this will go away when I merge/rebase your #480 right?

Reply:
The entire file is unnecessary with #480.  The command to create an instance of `runtime-images` will look like: `elyra-metadata install runtime-images --schema-name=runtime-image ...`